src/Abp/EntityHistory/EntitySnapshotManagerBase.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Abp.Dependency;
+using Abp.Domain.Entities;
+using Abp.Json;
+
+namespace Abp.EntityHistory
+{
+    public abstract class EntitySnapshotManagerBase : IEntitySnapshotManager, ITransientDependency
+    {
+        protected abstract Task<TEntity> GetEntityById<TEntity, TPrimaryKey>(TPrimaryKey id)
+            where TEntity : class, IEntity<TPrimaryKey>;
+
+        protected abstract IQueryable<EntityChange> GetEntityChanges<TEntity, TPrimaryKey>(TPrimaryKey id,
+            DateTime snapshotTime) where TEntity : class, IEntity<TPrimaryKey>;
+
+        public async Task<EntityHistorySnapshot> GetSnapshotAsync<TEntity>(int id, DateTime snapshotTime) where TEntity : class, IEntity<int>
```

Comment:
yes. an extensions will be good.

there are many examples in abp code base.

e.g.
https://github.com/aspnetboilerplate/aspnetboilerplate/blob/2144f552087d95f13ac7529e545ac36583f71681/src/Abp.Zero.Common/Runtime/Caching/AbpZeroCacheManagerExtensions.cs#L10-L13